lib/ruby_lsp/requests/semantic_highlighting.rb

```diff
@@ -105,6 +112,8 @@ def run
 
       sig { override.params(node: SyntaxTree::Call).void }
       def visit_call(node)
+        return super unless visible?(node, @range)
```

Comment:
This is a lot of repetition and ends up being brittle. If someone were to add a new method to visit and they forget to add this check, this would break.

I think we can do this for each node by overriding the `visit_child_nodes` which each visitor method is already aliased to.

I guess:
```ruby
def visit_child_nodes(node)
  return unless visible?(node, @range)

  super
end
```
should do the same thing?

Reply:
Sorry, I'd meant to comment, not approve 😅